From: Stefan Bruens <stefan.bruens@rwth-aachen.de>
Date: Fri, 20 Jul 2018 16:23:11 +0100
Subject: fix possible crash in qikea

Using dirp after closedir is invalid, so don't do it.
---
 source/libike/manager.file.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/source/libike/manager.file.cpp b/source/libike/manager.file.cpp
index 7fd3dc3..29382c4 100644
--- a/source/libike/manager.file.cpp
+++ b/source/libike/manager.file.cpp
@@ -109,13 +109,16 @@ bool _CONFIG_MANAGER::file_enumerate( CONFIG & config, int & index )
 		found++;
 	}
 
-	closedir( dirp );
 	if( dp == NULL )
+	{
+		closedir( dirp );
 		return false;
+	}
 
 	config.set_id( dp->d_name );
 	index++;
 
+	closedir( dirp );
 	return file_vpn_load( config );
 
 #endif
